components/pefa.py

```diff
@@ -0,0 +1,268 @@
+            'budget being implemented within a system of effective standards, processes, and internal controls, ensuring that resources are obtained and used as intended',
+        ]
+    ),
+    (
+        'pillar6_accounting_and_reporting',
+        [
+            'accurate and reliable records are maintained, and information is produced and disseminated at appropriate times to meet decision-making, management, and reporting needs',
+            'maintaining accurate and reliable records, producing and disseminating information at appropriate times to meet decision-making, management, and reporting needs',
+        ]
+    ),
+    (
+        'pillar7_external_audit',
+        [
+            'public finances being independently reviewed and presence of external follow-up on the implementation of recommendations for improvement by the executive',
+            'ensuring public finances is independently reviewed, and there is external follow-up on the implementation of recommendations for improvement by the executive',
+        ]
+    ),
+])
+
+
+def pefa_overall_figure(df, pov_df):
+    pillar_columns = [col for col in df.columns if col.startswith('pillar')]
+    overall_scores = df[pillar_columns].mean(axis=1)
+
+    overall_grades = overall_scores.map(_score_to_grade)
+
+    fig = make_subplots(specs=[[{"secondary_y": True}]])
+
+    fig.add_trace(
+        go.Scatter(
+            name="Poverty Rate",
+            x=pov_df.year,
+            y=pov_df.poor215,
+            mode="lines+markers",
+            line=dict(color="darkred", shape="spline", dash="dot"),
+            connectgaps=True,
+            hovertemplate=("%{x}: %{y:.2f}%"),
+        ),
+        secondary_y=True,
+    )
+
+    fig.add_trace(
+        go.Scatter(
+            name="PEFA Score",
+            x=df.year,
+            y=overall_scores,
+            mode="lines+markers",
+            marker_color="darkblue",
+            hovertemplate=("%{x}: %{y:.2f} (%{customdata})"),
+            customdata=overall_grades,
+        ),
+        secondary_y=False,
+    )
+
+    fig.update_xaxes(tickformat="d")
+    fig.update_yaxes(
+        title_text="Quality of Budget Institutions",
+        secondary_y=False,
+        tickvals=list(SCORE_MAPPING.keys()),
+        ticktext=list(SCORE_MAPPING.values()),
+        range=[0, 4.5],
+    )
+    fig.update_yaxes(
+        title_text="Poverty Rate (%)",
+        secondary_y=True,
+        range=[-10, 100],
+    )
+    title_text = "How did the overall quality of budget institutions change over time?"
+    wrapped_title = "<br>".join(textwrap.wrap(title_text, width=40))
+    fig.update_layout(
+        barmode="stack",
+        title={
+            'text': wrapped_title,
+            'font': {'size': 16},
+            'x': 0.5,
+            'y': 0.92,
+            'xanchor': 'center',
+        },
+        plot_bgcolor="white",
+        legend=dict(orientation="h", yanchor="bottom", y=1.03),
```

Comment:
x=0.5 would centre the legend under the legend title? It would also be good to have a little space between the legend title and the legend bar

Reply:
I assume you are talking about the heatmap legend, not the overall scatter plot? I've moved the heatmap title to the side to be consistent with the scatter plot and make the top less crowded. Let me know what you think.